contract/vault/soroban/scripts/vault_payload.py

```
@@ -0,0 +1,223 @@
+    return int.from_bytes(raw, "little", signed=False), cursor
+
+
+def read_i128(data: bytes, cursor: int) -> tuple[int, int]:
+    raw, cursor = read_exact(data, cursor, 16)
+    return int.from_bytes(raw, "little", signed=True), cursor
+
+
+def parse_u32_json(value: str) -> list[int]:
+    parsed = json.loads(value)
+    if not isinstance(parsed, list):
+        raise ValueError("expected a JSON array of u32 values")
```

Ruff (0.15.15)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)